ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/TraceBlock.java

```diff
@@ -66,19 +77,38 @@ protected BlockParameter blockParameter(final JsonRpcRequestContext request) {
 
   @Override
   protected Object resultByBlockNumber(
-      final JsonRpcRequestContext request, final long blockNumber) {
+      final JsonRpcRequestContext requestContext, final long blockNumber) {
     if (blockNumber == BlockHeader.GENESIS_BLOCK_NUMBER) {
       // Nothing to trace for the genesis block
       return emptyResult().getArrayNode();
     }
     LOG.trace("Received RPC rpcName={} block={}", getName(), blockNumber);
 
-    return getBlockchainQueries()
-        .getBlockchain()
-        .getBlockByNumber(blockNumber)
-        .map(block -> traceBlock(block, Optional.empty()))
-        .map(ArrayNodeWrapper::getArrayNode)
-        .orElse(null);
+    final ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
why creating a new executor every time and not using `EthScheduler` or what Vert.x provides to execute long running tasks?